src/core/stream_recv.c

```diff
@@ -966,13 +974,19 @@ QuicStreamRecvFlush(
 
         QUIC_STATUS Status = QuicStreamIndicateEvent(Stream, &Event);
 
-        Stream->Flags.ReceiveCallActive = FALSE;
+        //
+        // Clear the active receive flag.
+        //
+        RecvCompletionLength = InterlockedAnd64(
+            (int64_t*)&Stream->RecvCompletionLength,
+            ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG);
+        RecvCompletionLength &= ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG;
```

Comment:
```suggestion
        RecvCompletionLength =
            InterlockedAnd64(
                (int64_t*)&Stream->RecvCompletionLength,
                ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG) &
            ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG;
```

Reply:
Actually, we should use `InterlockedExchange64` since our goal is to clear the whole thing anyways, so we don't need to do another interlocked operation down below to subtract.
```suggestion
        RecvCompletionLength =
            InterlockedExchange64(
                (int64_t*)&Stream->RecvCompletionLength,
                0) &
            ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG;
```
You might have to define this in quic_platform_posix.h to call `__sync_lock_test_and_set`.

Then, you can completely remove the `BufferLength` variable (using `RecvCompletionLength` directly) and the interlocked function below.